File: VSTTwain/Examples/CSharp/TwainAdvancedDemo/Dialogs/SaveSettings/PdfSaveSettingsForm.cs
using System;
using System.Windows.Forms;
using Vintasoft.Twain;
using Vintasoft.Twain.ImageEncoders;

namespace TwainAdvancedDemo
{
	public partial class PdfSaveSettingsForm : Form
    {

        #region Fields & properties

        bool _saveAllImages = false;
        public bool SaveAllImages
        {
            get { return _saveAllImages; }
        }
        
        bool _multiPage = true;
		public bool MultiPage
		{
			get { return _multiPage; }
		}

		bool _pdfACompatible = true;
		public bool PdfACompatible
		{
			get { return _pdfACompatible; }
		}

		string _pdfAuthor = string.Empty;
		public string PdfAuthor
		{
			get { return _pdfAuthor; }
		}

		string _pdfTitle = string.Empty;
		public string PdfTitle
		{
			get { return _pdfTitle; }
		}

		PdfImageCompression _compression = PdfImageCompression.Auto;
		public PdfImageCompression Compression
		{
			get { return _compression; }
		}
		
		int _jpegQuality = 90;
		public int JpegQuality
		{
			get { return _jpegQuality; }
        }

        #endregion


        #region Constructor

        public PdfSaveSettingsForm(bool isFileExist)
		{
			InitializeComponent();

			if (!isFileExist)
			{
				createNewDocumentRadioButton.Checked = true;
				addToDocumentRadioButton.Enabled = false;
			}
        }

        #endregion


        #region Methods

        private void okButton_Click(object sender, EventArgs e)
		{
            _saveAllImages = saveAllImagesRadioButton.Checked;

			_multiPage = addToDocumentRadioButton.Checked;
			_pdfACompatible = pdfACompatibleCheckBox.Checked;
			_pdfAuthor = pdfAuthorTextBox.Text;
			_pdfTitle = pdfTitleTextBox.Text;

			if (noneCompressionRadioButton.Checked)
                _compression = PdfImageCompression.None;
			else if (ccittCompressionRadioButton.Checked)
                _compression = PdfImageCompression.CcittFax;
			else if (lzwCompressionRadioButton.Checked)
                _compression = PdfImageCompression.LZW;
			else if (jpegCompressionRadioButton.Checked)
			{
				_compression = PdfImageCompression.JPEG;
				_jpegQuality = (int)jpegQualityNumericUpDown.Value;
			}
			else if (zipCompressionRadioButton.Checked)
                _compression = PdfImageCompression.ZIP;
			else if (autoCompressionRadioButton.Checked)
                _compression = PdfImageCompression.Auto;

			DialogResult = DialogResult.OK;
		}

		private void EnableJpegCompressionQuality(object sender, EventArgs e)
		{
			gbJpegCompression.Enabled = true;
		}

		private void DisableJpegCompressionQuality(object sender, EventArgs e)
		{
			gbJpegCompression.Enabled = false;
        }

        #endregion

    }
}

File: VSTTwain/Examples/VB.NET/TwainAdvancedDemo/Dialogs/SaveSettings/PdfSaveSettingsForm.vb
Imports System.Windows.Forms
Imports Vintasoft.Twain
Imports Vintasoft.Twain.ImageEncoders

Partial Public Class PdfSaveSettingsForm
    Inherits Form

#Region "Fields & properties"

    Private _saveAllImages As Boolean = False
    Public ReadOnly Property SaveAllImages() As Boolean
        Get
            Return _saveAllImages
        End Get
    End Property

    Private _multiPage As Boolean = True
    Public ReadOnly Property MultiPage() As Boolean
        Get
            Return _multiPage
        End Get
    End Property

    Private _pdfACompatible As Boolean = True
    Public ReadOnly Property PdfACompatible() As Boolean
        Get
            Return _pdfACompatible
        End Get
    End Property

    Private _pdfAuthor As String = String.Empty
    Public ReadOnly Property PdfAuthor() As String
        Get
            Return _pdfAuthor
        End Get
    End Property

    Private _pdfTitle As String = String.Empty
    Public ReadOnly Property PdfTitle() As String
        Get
            Return _pdfTitle
        End Get
    End Property

    Private _compression As PdfImageCompression = PdfImageCompression.Auto
    Public ReadOnly Property Compression() As PdfImageCompression
        Get
            Return _compression
        End Get
    End Property

    Private _jpegQuality As Integer = 90
    Public ReadOnly Property JpegQuality() As Integer
        Get
            Return _jpegQuality
        End Get
    End Property

#End Region


#Region "Constructor"

    Public Sub New(ByVal isFileExist As Boolean)
        InitializeComponent()

        If Not isFileExist Then
            createNewDocumentRadioButton.Checked = True
            addToDocumentRadioButton.Enabled = False
        End If
    End Sub

#End Region


#Region "Methods"

    Private Sub okButton_Click(ByVal sender As Object, ByVal e As EventArgs)
        _saveAllImages = saveAllImagesRadioButton.Checked

        _multiPage = addToDocumentRadioButton.Checked
        _pdfACompatible = pdfACompatibleCheckBox.Checked
        _pdfAuthor = pdfAuthorTextBox.Text
        _pdfTitle = pdfTitleTextBox.Text

        If noneCompressionRadioButton.Checked Then
            _compression = PdfImageCompression.None
        ElseIf ccittCompressionRadioButton.Checked Then
            _compression = PdfImageCompression.CcittFax
        ElseIf lzwCompressionRadioButton.Checked Then
            _compression = PdfImageCompression.LZW
        ElseIf jpegCompressionRadioButton.Checked Then
            _compression = PdfImageCompression.JPEG
            _jpegQuality = CInt(Math.Truncate(jpegQualityNumericUpDown.Value))
        ElseIf zipCompressionRadioButton.Checked Then
            _compression = PdfImageCompression.ZIP
        ElseIf autoCompressionRadioButton.Checked Then
            _compression = PdfImageCompression.Auto
        End If

        DialogResult = DialogResult.OK
    End Sub

    Private Sub EnableJpegCompressionQuality(ByVal sender As Object, ByVal e As EventArgs)
        gbJpegCompression.Enabled = True
    End Sub

    Private Sub DisableJpegCompressionQuality(ByVal sender As Object, ByVal e As EventArgs)
        gbJpegCompression.Enabled = False
    End Sub

#End Region

End Class
